docs/shared-content-source/docs/modules/develop/pages/akka-streamlet-utilities.adoc

```diff
@@ -106,6 +107,68 @@ include::{cloudflow-examples-version}@docsnippets:ROOT:example$akkastreams-scala
 
 The above Scala example creates a route that will handle put requests where the entity contains `Data`, which is written to the outlet using the `WritableSinkRef`.
 
+== GrpcServerLogic
+
+=== Use case
+
+An `GrpcServerLogic` can be used to handle gRPC requests.
+You need to extend your streamlet from `AkkaServerStreamlet`
+so that Cloudflow will expose an HTTP endpoint in Kubernetes.
+
+Additionally, you need to enable HTTP/2 by adding the following
+to your configuration:
+
+[source,hocon]
+----
+akka.http.server.preview.enable-http2 = on
```

Comment:
Would it be possible to 'automate this' so that the user does not have to do it? We do some configuration changes in AkkaStreamlet for cluster config for instance: https://github.com/lightbend/cloudflow/blob/master/core/cloudflow-akka/src/main/scala/cloudflow/akkastream/AkkaStreamlet.scala#L45-L85

Reply:
Since we're in charge of creating the ActorSystem this is indeed possible. It does raise some questions: should it be possible for the user to override this? If yes, Play and Lagom also have this 'problem' (and IIRC have different ways to solve it), should we introduce a 3rd approach or follow one of those?

Reply:
In this case, a streamlet always runs alone (in a pod /container) somewhere. The user indicates by using `GrpcServerLogic` what they want to do, server gRPC, which as I understand, requires http2. Not setting this would break something, correct? Also, we have a config system in place for cloudflow, they actually can already override anythingthey like at deployment if they really wanted to. Please see https://cloudflow.io/docs/current/develop/cloudflow-configuration.html (they could `set cloudflow.runtimes.akka.config.akka.http.server.preview.enable-http2 = off` in a config file and do `kubectl cloudflow deploy app.json --conf my.conf` although I don't think that makes sense here, though I know too little about gRPC at this point to say @raboof.

Reply:
Since this means changing some relatively low-level cloudflow infrastructure, I think it'd be good to do this in a separate PR

Reply:
Sure no problem!

Reply:
filed https://github.com/lightbend/cloudflow/issues/666